src/WebJobs.Script.WebHost/Security/Authentication/Keys/AuthenticationLevelHandler.cs

```diff
@@ -44,6 +44,17 @@ protected override async Task<AuthenticateResult> HandleAuthenticateAsync()
             // Get the authorization level for the current request
             (string name, AuthorizationLevel requestAuthorizationLevel) = await GetAuthorizationKeyInfoAsync(Context.Request, _secretManagerProvider);
 
+            List<ClaimsIdentity> claimsIdentities = new List<ClaimsIdentity>();
+
+            if (_isEasyAuthEnabled)
+            {
+                ClaimsIdentity easyAuthIdentity = Context.Request.GetAppServiceIdentity();
+                if (easyAuthIdentity != null)
+                {
+                    claimsIdentities.Add(easyAuthIdentity);
+                }
+            }
+
             if (requestAuthorizationLevel != AuthorizationLevel.Anonymous)
```

Comment:
`HttpTrigger_Identities_Succeeds` and `HttpTrigger_Identities_AnonymousAccessSucceeds` both use the `HttpTrigger-Identities` test function. Now that its `authLevel` is `anonymous`, does `HttpTrigger_Identities_Succeeds` still execute this code block?

Reply:
Good point. Unfortunately, spinning up a new sample just to change `authLevel` seems like it would start polluting the "samples" directory, and making the "authLevel" customization by using a binding expression seems to limit the utility of the sample as an example for how to use the ClaimsPrincipal binding. Maybe it would be better to test this scenario at a lower level than these E2E tests.

Reply:
Disregard the previous message, it's been a while since I have touched this code.

This code path actually is touched, as `requestAuthorizationLevel` is actually the authorization level of the key provided to the current HTTP request, i.e. what level the request is authenticated for. The fact that the function is set to only require `anonymous` isn't actually taken into account until the `AuthLevelAuthorizationHandler.cs` checks whether the request's authentication level is greater than or equal what the Function requires.